apiserver/storage/storage_test.go

```diff
@@ -343,3 +343,21 @@ func (s *storageSuite) TestDestroy(c *gc.C) {
 		destroyStorageInstanceCall,
 	})
 }
+
+func (s *storageSuite) TestAttach(c *gc.C) {
+	results, err := s.api.Attach(params.StorageAttachParams{[]params.StorageAttachParam{
+		{StorageTag: "storage-data-0", EntityTag: "unit-mysql-0"},
+		{StorageTag: "storage-data-0", EntityTag: "machine-0"},
+		{StorageTag: "volume-0", EntityTag: "unit-mysql-0"},
+	}})
+	c.Assert(err, gc.ErrorMatches, "attaching storage not implemented")
+	c.Assert(results.Results, gc.HasLen, 3)
+	c.Assert(results.Results, jc.DeepEquals, []params.ErrorResult{
+		{Error: nil},
+		{Error: &params.Error{Message: `entity kind "machine" for storage attachment not valid`}},
+		{Error: &params.Error{Message: `"volume-0" is not a valid storage tag`}},
+	})
+	s.assertCalls(c, []string{
+		getBlockForTypeCall, // Change
```

Comment:
it's a "change" block check, same as all the other tests.

we should be using testing.Stub, and checking the value. I'll look at moving it all over, if it's not too much work.